robosuite/robots/legged_robot.py

```diff
@@ -188,6 +189,17 @@ def control(self, action, policy_step=False):
             applied_action_high = self.sim.model.actuator_ctrlrange[self._ref_actuators_indexes_dict[part_name], 1]
             actuator_indexes = self._ref_actuators_indexes_dict[part_name]
             actuator_gears = self.sim.model.actuator_gear[actuator_indexes, 0]
+
+            part_controllers = self.composite_controller.get_controller(part_name)
+            if (
+                isinstance(part_controllers, JointPositionController)
+                or isinstance(part_controllers, JointVelocityController)
+                or isinstance(part_controllers, JointTorqueController)
+            ):
+                # select only the joints that are actuated
+                actuated_joint_indexes = self._ref_actuator_to_joint_id[actuator_indexes]
+                applied_action = applied_action[actuated_joint_indexes]
+
```

Comment:
would this logic apply more generally to other robots? other than legged_robot.py

also, why is the logic only applied for JointPositionController/JointVelocityController/JointTorqueController? Is it because these are the controllers where we directly output joint targets?

Reply:
Yes, this logic should apply to all robots.

Yes. The JointController generates control signals for all joints, while the GripperController only produces control signals for actuators. This difference explains why we need these conditional statements. I'm uncertain about the behavior of other controllers. 
Also I was thinking that the GripperController seems essentially similar to JointController but focuses only on actuator-driven joints. If so, we could potentially unify their implementation since their fundamental problem if the same: align joint with actuator actions.